packages/dd-trace/src/encode/tags-processors.js

```diff
@@ -38,11 +38,12 @@ function truncateToLength (value, maxLength) {
   return value
 }
 
-function truncateSpan (span) {
+// normally the agent truncates the resource and parses it in certain scenarios (e.g. SQL Queries)
+function truncateSpan (span, truncateResource) {
   return fromEntries(Object.entries(span).map(([key, value]) => {
     switch (key) {
       case 'resource':
-        return ['resource', truncateToLength(value, MAX_RESOURCE_NAME_LENGTH)]
+        return truncateResource ? ['resource', truncateToLength(value, MAX_RESOURCE_NAME_LENGTH)] : [key, value]
```

Comment:
maybe this is more concise? 
```suggestion
        return ['resource', shouldTruncateResourceName  ? truncateToLength(value, MAX_RESOURCE_NAME_LENGTH)] : value]
```